src/cache/cache_las.c

```diff
@@ -1101,65 +1110,85 @@ __wt_las_sweep(WT_SESSION_IMPL *session)
           cursor->get_value(cursor, &durable_timestamp, &prepare_state, &upd_type, &las_value));
 
         /*
-         * Check to see if the page or key has changed this iteration,
-         * and if they have, setup context for safely removing obsolete
-         * updates.
-         *
-         * It's important to check for page boundaries explicitly
-         * because it is possible for the same key to be at the start
-         * of the next block. See WT-3982 for details.
+         * Expect an update entry with:
+         *	1. not in a prepare locked state
+         *	2. durable timestamp as not max timestamp.
+         *	3. for an in-progress prepared update, durable
+         *	timestamp should be zero.
+         *	4. no restriction on durable timestamp value
+         *	for other updates.
          */
-        if (las_btree_id != saved_btree_id || saved_key->size != las_key.size ||
-          memcmp(saved_key->data, las_key.data, las_key.size) != 0) {
-            /* If we've examined enough entries, give up. */
-            if (cnt == 0)
-                break;
+        WT_ASSERT(session, prepare_state != WT_PREPARE_LOCKED && durable_timestamp != WT_TS_MAX &&
+            (prepare_state != WT_PREPARE_INPROGRESS || durable_timestamp == 0));
 
-            saved_btree_id = las_btree_id;
-            WT_ERR(__wt_buf_set(session, saved_key, las_key.data, las_key.size));
+        WT_ASSERT(
+          session, (prepare_state == WT_PREPARE_INPROGRESS || durable_timestamp >= las_timestamp));
 
-            /*
-             * Expect an update entry with:
-             *	1. not in a prepare locked state
-             *	2. durable timestamp as not max timestamp.
-             *	3. for an in-progress prepared update, durable
-             *	timestamp should be zero.
-             *	4. no restriction on durable timestamp value
-             *	for other updates.
-             */
-            WT_ASSERT(session, prepare_state != WT_PREPARE_LOCKED &&
-                durable_timestamp != WT_TS_MAX &&
-                (prepare_state != WT_PREPARE_INPROGRESS || durable_timestamp == 0));
+        /*
+         * Check to see if the page or key has changed this iteration, and if they have, setup
+         * context for safely removing obsolete updates.
+         */
+        remove_entry = false;
+        if (prev_entry_can_remove) {
+            if (las_btree_id != saved_btree_id || saved_key->size != las_key.size ||
+              memcmp(saved_key->data, las_key.data, las_key.size) != 0) {
+                remove_entry = true;
+                saved_btree_id = las_btree_id;
+                WT_ERR(__wt_buf_set(session, saved_key, las_key.data, las_key.size));
+            } else {
+                if (upd_type != WT_UPDATE_MODIFY)
+                    remove_entry = true;
+                else
+                    skip_update_modify_cnt++;
+            }
```

Comment:
I'd write
```
if (prev_entry_can_remove) {
    if (las_btree_id != saved_btree_id || saved_key->size != las_key.size ||
      memcmp(saved_key->data, las_key.data, las_key.size) != 0) {
        remove_entry = true;
        saved_btree_id = las_btree_id;
        WT_ERR(__wt_buf_set(session, saved_key, las_key.data, las_key.size));
    } else {
        if (upd_type != WT_UPDATE_MODIFY)
            remove_entry = true;
        else
            skip_update_modify_cnt++;
    }   
}   
```
as:
```
    }
    else if (upd_type != WT_UPDATE_MODIFY)  
        remove_entry = true;
    else
        skip_update_modify_cnt++;
}   
```
there's no need for an additional indented block?

Reply:
Fixed.